crates/ruff_linter/src/rules/flake8_pyi/rules/unrecognized_platform.rs

```diff
@@ -130,7 +130,7 @@ pub(crate) fn unrecognized_platform(checker: &mut Checker, test: &Expr) {
             if !matches!(value.as_str(), "linux" | "win32" | "cygwin" | "darwin") {
```

Comment:
This is an example where... we can probably do this without the concatenation? Imagine if we had a custom matcher that worked on implicit concatenations (i.e., a vector of strings). \cc @burntsushi in case you have obvious ideas here.

Reply:
I think the closest "out of the box" experience you can get is [stream searching with `aho-corasick`](https://docs.rs/aho-corasick/latest/aho_corasick/struct.AhoCorasick.html#method.stream_find_iter), but you need to provide a `std::io::Read` impl. Mildly annoying but doable in this context.

But I would say to write the simple/obvious code for now, and if this ends up popping up on a profile then we can revisit it and do something bespoke here or use `aho-corasick`.

Reply:
I think I would prefer to avoid the complexity for such simple cases but I do understand that this could be useful in similar cases where the strings are long.